psq/datastore_storage.py

```diff
@@ -26,6 +27,13 @@
 
 DATASTORE_KIND_PREFIX = 'psq'
 
+_RETRY = retry(
+    stop_max_attempt_number=5,
+    wait_exponential_multiplier=1000,
+    wait_exponential_max=10000,
+    retry_on_exception=lambda e: not isinstance(e, KeyboardInterrupt)
```

Comment:
Ah, now I can see. We should only retry on `google.cloud.exceptions.GoogleCloudError`, or maybe even just a subset of those.